source/scripts/RecipePage.js

```diff
@@ -0,0 +1,230 @@
+    {
+        content: "\\2714"; /* tick sign */
+        font-size: 15px;
+    }
+  
+    #ingredients-list > ul > ol > label{
+      color: black;
+    }
+    /*-----------------------------------*/
+    
+    #instructions > ol > li{
+      color: black;
+    }
+  
+    #ingredients-list{
+        /* text-align: center; */
+        font-style: normal;
+        font-weight: normal;
+  
+    }
+  
+    #ingredients-list > button{
+      border: 0;
+      background-color: inherit;
+      font: inherit;
+      text-align: center;
+      padding: 20px;
+    }
+  
+    #ingredients-list> button:hover{
+      cursor: pointer;
+      background: #eee;
+    }
+      `;
+
+
+    this.shadowRoot.append(style, container);
+  }
+
+  set data(data) {
+    if (data == null) {
+      console.log("Error: no data exists");
+      return;
+    }
+
+    this.shadowRoot.querySelector("article").innerHTML = `
+      <section class="middle">
+        <header class="header">
+          <h1></h1>
+            <img id="bookmark-icon" onclick="toggleBookMark(this)" src="./img/icons/bookmark-empty.svg"" width="56" height="56">
+         </header>
+          <h1></h1>
+          <div id="ingredients-list">
+              <h3>INGREDIENTS</h3>
+              <ul style="list-style-type: none;">
+              </ul>
+              <button id="clear-checkboxes" onclick="clearCheckBoxes()">CLEAR CHECKBOXES</button>
+          </div>
+          <div id="instructions">
+              <h3>INSTRUCTIONS</h3>   
+              <ol>
+              </ol>
+          </div>
+      </section>
+      `;
+
+    // Header - title
+
+    this.shadowRoot.querySelector(".header > h1").innerHTML = data["title"];
+
+    //get ingredient list
+    const ingredients = getIngredients(data);
+    ingredients.forEach(ingredient => {
+      const checkbox = document.createElement("input");
+      const label = document.createElement("label");
+      const ol = document.createElement("ol");
+      checkbox.type = "checkbox";
+      checkbox.classList.add("ingredients-custom-checkbox");
+      label.innerText = ingredient;
+      ol.appendChild(checkbox);
+      ol.appendChild(label);
+      this.shadowRoot.querySelector("#ingredients-list > ul").appendChild(ol);
+    });
+
+    // <-- instruction -->
+    const instrucstions = getInstructions(data);
+    instrucstions.forEach(element => {
+      const li = document.createElement("li");
+      li.innerHTML = element;
+      this.shadowRoot.querySelector("#instructions > ol").appendChild(li);
+    });
+  }
+}
+
+
+// Helper functions
+/**
+ * Extract the ingredients from data
+ * @param {Object} data JSON
+ * @returns {Array} return a list of ingredients
+ */
+function getIngredients(data) {
+  const steps = data["analyzedInstructions"][0]["steps"];
+  let list = [];
+  let index = 0;
+  steps.forEach((step) => {
+    let ingredients = step["ingredients"];
+    console.log('ingredients', ingredients);
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8414136)
```suggestion
    console.log("ingredients", ingredients);
```